plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+                    "mac_address": "90:e9:5e:03:f3:40"
+                }
+            ]
+        Description:
+            Checks the existence of a Device details in Cisco Catalyst Center by querying the
+            'get_device_list' function in the 'devices' family. Based on the response need to 
+            pass the ap_ethernet_mac_address to the another 'get_access_point_configuration' function
+            in the 'wireless' family to get AP configuration to check the input data with current
+            config data and return above resoponse.
+        """
+        accesspoint_exists = False
+        current_configuration = {}
+        response = None
+        input_param = {}
+        if input_config.get("mac_address") is not None and input_config.get("mac_address") != "":
+            input_param["macAddress"] = input_config["mac_address"]
+        elif input_config.get("management_ip_address") is not None and \
+            input_config.get("management_ip_address") != "":
+            input_param["managementIpAddress"] = input_config["management_ip_address"]
+        elif input_config.get("hostname") is not None and input_config.get("hostname") != "":
+            input_param["hostname"] = input_config["hostname"]
+
+        if not input_param:
+            self.log("Required param mac_address, management_ip_address or hostname is not in playbook config",
+                      "ERROR")
+            return (accesspoint_exists, current_configuration)
+
+        try:
+            response = self.dnac._exec(
+                family="devices",
+                function='get_device_list',
+                op_modifies=True,
+                params=input_param,
+            )
+        except Exception as e:
+            self.log("The provided device '{0}' is either invalid or not present in the Cisco Catalyst Center."\
+                     .format(str(input_param) + str(e)), "WARNING")
+        if response:
+            response = response.get("response")
+            response = self.camel_to_snake_case(response)
+            device_fields = self.payload.get("config")[0].get("device_fields")
+            if device_fields is None or device_fields == "" or device_fields == "all":
+                self.payload["device_list"] = response[0]
+            else:
+                self.payload["device_list"]=self.data_frame(device_fields,response)
+            self.log("Received API response from 'get_device_list': {0}"\
+                     .format(str(self.payload["device_list"][0])), "DEBUG")
+
+            ap_response = None
+            input_param = {}
+            ap_ethernet_mac_address = response[0]["ap_ethernet_mac_address"]
+            input_param["key"] = ap_ethernet_mac_address
+            try:
+                ap_response = self.dnac._exec(
+                    family="wireless",
+                    function='get_access_point_configuration',
+                    params=input_param,
+                )
```

Comment:
Don't invoke multiple API in a single function, make the modular function.

Reply:
moved to another function for AP config data.